### PR TITLE
 Rework the property factory to allow stream specific factories, and tweak the logic for deciding whether a property value should be padded.

### PR DESCRIPTION
**Pull request by Numpsy**
A few experiments based on #99 and the comments in #98 - 

Instead of having a single, singleton instance of a property factory, which is accessed in several places, instead pass an instance of the factory around, and decide which factory to use at a higher level.

When creating a property via the factory, pass the identifier of the property into the factory, so that it can make property specific choices in which property type to use when needed (mainly to use VtUnalignedString for a couple of string properties, rather than VtString)

Add an additional property factory for the DocumentSummaryInformation stream which has specific knowledge of a couple of unaligned string properties.

Also added the unit test and extra test file from #99 / #98 as a test case for a file that isn't currently read correctly.

### CODE REVIEW

**Comment by ironfede** [comment]
Many thanks @Numpsy :I don't think that a solution for the underlying issue can be "super-clean" from a code perspective since the issue derives from a tangled specifc in itself. So I think you should go on on with your experiments towards a fully working solution and only after that starting to think about a clean-up refactoring. I'm sorry for being not-active on this threads but I'm currently out of time to better follow all discussions and evolution of code. Thank you anyway for your developments.

File: sources/OpenMcdf.Extensions/OLEProperties/PropertyFactory.cs [hunk 1]
```diff
@@ -1,35 +1,20 @@
...
 using System.Text;
 using System.IO;
-using System.Threading;
 
 namespace OpenMcdf.Extensions.OLEProperties
 {
-    internal class PropertyFactory
+    internal abstract class PropertyFactory
     {
-        private static ThreadLocal<PropertyFactory> instance
```
Comment (Numpsy) [inline sources/OpenMcdf.Extensions/OLEProperties/PropertyFactory.cs:12]: As it stands, the factory class is stateless, so I don't know why it needs to be ThreadLocal? (if not, that change could be broken out into a separate PR to make things simpler here)

**Comment by Numpsy** [comment]
I'm going to try running some property read/write tests on a set of test documents at work to see if it shakes out any more bugs, and then see what to do with more changes after that.

**Comment by Numpsy** [comment]
I'll have another look now the other changes are merged (I realised whilst testing those that the current code here is wrong, as it tries to use DocumentSummaryInfoPropertyFactory for user defined properties, which doesn't work because you can't determine the behavior based on property identifier for that)

**Comment by Numpsy** [comment]
Rebased onto the other recent changes, and tweaked a little to minimze the changes (the changes in TypedPropertyValue.cs look bigger than they really are due to bracing / indentation changes)
